src/Storages/DataLakes/IStorageDataLake.h

```diff
@@ -22,15 +22,15 @@ class IStorageDataLake : public Storage
     using Configuration = typename Storage::Configuration;
 
     template <class ...Args>
-    explicit IStorageDataLake(const Configuration & configuration_, ContextPtr context_, Args && ...args)
-        : Storage(getConfigurationForDataRead(configuration_, context_), context_, std::forward<Args>(args)...)
+    explicit IStorageDataLake(const Configuration & configuration_, ContextPtr context_, bool attach, Args && ...args)
```

Comment:
I don't quite understand why we need `attach` flag in the constructor, I see that we set it alwasy to `true` in `ITableFunctionDataLake.h`.
We can just add `throw_on_error` flag to `getConfigurationForDataRead` and set it to `false` when it's called from constructor.

Reply:
I think it's ok not to throw not only in case of attach, but always on storage creation and throw only when we read from it.

Reply:
> I think it's ok not to throw not only in case of attach, but always on storage creation and throw only when we read from it.

I think we should not silence the error on create query, it is done this way for several table engines already.